src/main/java/com/google/api/codegen/discovery/ViewModelProvider.java

```
@@ -0,0 +1,191 @@
+    docs.put(outputRoot + "/" + surfaceDoc.outputPath(), doc);
+    return docs;
+  }
+
+  /**
+   * Applies sampleConfigOverrides to sampleConfig.
+   *
+   * If sampleConfigOverrides is null, sampleConfig is returned as is.
+   */
+  private static SampleConfig override(SampleConfig sampleConfig, JsonNode sampleConfigOverrides) {
+    // We use JSON merging to facilitate this override mechanism:
+    // 1. Convert the SampleConfig into a JSON tree.
+    // 2. Convert the overrides file into a JSON tree with arbitrary schema.
+    // 3. Overwrite object fields of the SampleConfig tree where field names match.
+    // 4. Convert the modified SampleConfig tree back into a SampleConfig.
+    if (sampleConfigOverrides != null) {
+      ObjectMapper mapper = new ObjectMapper();
+      JsonNode tree = mapper.valueToTree(sampleConfig);
+      merge((ObjectNode) tree, (ObjectNode) sampleConfigOverrides);
+      try {
+        return mapper.treeToValue(tree, SampleConfig.class);
+      } catch (Exception e) {
+        throw new RuntimeException("failed to parse config to node: " + e.getMessage());
+      }
+    }
+    return sampleConfig;
+  }
+
+  /**
+   * Overwrites the fields of tree that intersect with those of overrideTree.
+   *
+   * Values of tree are modified, and values of overrideTree are not modified.
+   * The merge process loops through the fields of tree and replaces non-object
+   * values with the corresponding value from overrideTree if present. Object
+   * values are traversed recursively to replace sub-properties.
+   *
+   * @param tree the original JsonNode to modify.
+   * @param overrideTree the JsonNode with values to overwrite tree with.
+   */
+  private static void merge(ObjectNode tree, ObjectNode overrideTree) {
+    Iterator<String> fieldNames = overrideTree.fieldNames();
+    while (fieldNames.hasNext()) {
+      String fieldName = fieldNames.next();
+      JsonNode primaryValue = tree.get(fieldName);
+      if (primaryValue == null) {
```

Right, that's my point. Why not make this an error (exception), to explicitly point out that the user is trying to override a nonexistent node? What's the thinking behind ignoring and silently continuing?